command/run.go

```diff
@@ -197,6 +198,10 @@ func (r *Reader) read() {
 			}
 			if err != nil {
 				if err == io.EOF {
+					if errors.Is(r.ctx.Err(), context.Canceled) {
+						r.err = r.ctx.Err()
+						return
```

Comment:
does not need early-return here. the following line already returns with an error.